OneCommunityGlobal/HGNRest
# Xiao add toFixed(2) to round hour numbers to two digits

# Description
(PRIORITY LOW) Jae: Round hours numbers to two digits
a. Below are examples of an email for going over hours on a project and for logging too few hours in a week. I’d like these both rounded to the hundredths place so they look cleaner and more professional. 
b. Same for everywhere else hours are shown
![image](https://github.com/OneCommunityGlobal/HGNRest/assets/44281916/3143d4b4-302c-4c29-a123-0ab161277c1d)

## Main changes explained:
- Added .toFixed(2) to parameter hoursLogged for function notifyTaskOvertimeEmailBody in TimeEntryController.js
- Added .toFixed(2) to parameter timeSpent for function assignBlueSquareForTimeNotMet in userHelper.js

## How to test:
The following steps demonstrate how to test function notifyTaskOvertimeEmailBody in TimeEntryController.js:
1. Check into current branch
2. To send email using Gmail API using OAuth2 and nodemailer, kindly follow the first 10 mins of this tutorial (https://www.youtube.com/watch?v=-rcRf7yswfM&t=10s) and setup a test email id and fill below variable values from the above setup in the .env file of the backend:
REACT_APP_EMAIL =
REACT_APP_EMAIL_CLIENT_ID =
REACT_APP_EMAIL_CLIENT_SECRET =
REACT_APP_EMAIL_REFRESH_TOKEN =
REACT_APP_EMAIL_CLIENT_REDIRECT_URI = https://developers.google.com/oauthplayground
sendEmail = TRUE
3. Go to file HGNRest/src/cronjobs/userProfileJobs.js, comment out the existing code, and fill the code like this:

const { CronJob } = require('cron');
const timeEntrycontroller = require('../controllers/timeEntryController')();

const userProfileJobs = () => {
  const test1 = new CronJob(
    '55 * * * *', // at minute 55. You can change the minute whenever you want to test
    async () => {
      await timeEntrycontroller.checkTaskOvertime(); // the function you want to test
    },
    null,
    false,
    'America/Los_Angeles',
  );
  test1.start();
};

module.exports = userProfileJobs;

Relevant documents: 
https://docs.google.com/document/d/1j92U7Vr14gQLehr1e_gkMdqWcOA37WGxNLv9yTa96CY/edit
https://docs.google.com/document/d/1By_Ls6RGbesMgZvRzU2zIPIJtLBKMgQjabTUH2ZM8kE/edit?pli=1#heading=h.ubnt1hjxaooh
5. Do `npm install`, `npm run build`, and `npm start` to run this PR locally
6. Log as admin user, go to Other Links -> Projects -> AAA USE THIS PROJECT -> WBS -> JAE'S WBS - USE THIS, create a new task, assign a test user to the task, and input estimated hours
7. Go to the assigned user's dashboard by clicking the dot to the left of the user's name from the leaderboard, click the yellow button "Add Time Entry for user", and log tangible time entry GREATER THAN estimated hours for the task assigned to this user (the time entry should include minutes and will become infinite decimals when converting to hours)
8. Check the email you used to register for Gmail API in Step 2, verify the Hours Logged is rounded to two decimals
![Pasted Graphic 6](https://github.com/OneCommunityGlobal/HGNRest/assets/44281916/64339718-5a4c-4a8c-80b6-1c433dde8ae4)

## Note:
I can't find a way to test function assignBlueSquareForTimeNotMet in userHelper.js, but the logic is the same, so the hours should be rounded there as well. If you find a way to test this function as well, please let me know.
